### DISCUSSION TITLE
Why the result of cylinder heat conduction simulated from MOOSE is very different from the theory resolution?

### DISCUSSION THREAD

**Discussion by Wang-Yihu** [idaholab/moose#28093] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [x] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question
The theory resolution for heat conduction problem in a cylinder with inner heat source:
![image](https://github.com/idaholab/moose/assets/53902666/a571c7a1-1230-4792-ae31-f85fe74aa467)
Imagine a heat conduction problem in a cylinder with inner heat source:
q = 2240000, T_w = 300, R=0.5, k=200, L = 10. In heat conduction theory, the temperature distribution will be 𝑇 = −(q/4𝑘)*𝑟^2 + T_w + (q*𝑅^2)/(4𝑘). The maximum temperature is certainly in the middle, when r = 0, T will be 1000.
So I edit 2DCylindricalTube.i:
```
[Mesh]
 [gmg]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 5
    ny = 10
    xmin = 0.0
    xmax = 0.5
    ymin = 0.0
    ymax = 10.0
  []
  coord_type = RZ
  rz_coord_axis = Y
[]

[Variables]
  [temp]
    family = LAGRANGE
    initial_condition = 300.0
  []
[]

[Kernels]
  [heat_1]
    type = HeatConduction
    variable = temp
  []
  [heat_2]
    type = HeatSource
    variable = temp
    function = 2240000.0
  []
[]

[Materials]
  [daore_steel]
    type = HeatConductionMaterial
    temp = temp
    thermal_conductivity = 200.0
  []
[]

[Problem]
  type = FEProblem
[]

[Executioner]
  type = Steady
  solve_type = PJFNK
[]

[Postprocessors]
  [output_heat]
    type = SideDiffusiveFluxIntegral
    variable = temp
    boundary = right
    diffusivity = thermal_conductivity
  []
  [output_Tmax]
    type = NodalMaxValue
    variable = temp
  []
[]

[BCs]
  [rightbc]
    type = DirichletBC
    variable = temp
    boundary = right
    value = 300.0
  []
[]

[Outputs]
  exodus = true
[]
```
However, the result is
![image](https://github.com/idaholab/moose/assets/53902666/d9944ff1-74c0-40fb-ae1e-1e83fdc78340)
Tmax does not equal to 1000. 
So I also edit 3DCylindricalTube.i:
```
[Mesh]
  [accg]
    type = AdvancedConcentricCircleGenerator
    ring_radii = '0.5'
    ring_intervals = '5'
    ring_block_ids = '15 20'
    ring_block_names = 'inner_tri inner'
    external_boundary_id = 100
    external_boundary_name = 'right'
    num_sectors = 40
    preserve_volumes = False
  []
  
  [extrude1]
    type = AdvancedExtruderGenerator
    input = accg
    heights = 10.0
    num_layers = 10
    direction = '0 0 1'
  []
[]

[Variables]
  [temp]
    family = LAGRANGE
    initial_condition = 300.0
  []
[]

[Kernels]
  [heat_1]
    type = HeatConduction
    variable = temp
  []
  [heat_2]
    type = HeatSource
    variable = temp
    function = 2240000.0
  []
[]

[Materials]
  [daore_steel]
    type = HeatConductionMaterial
    temp = temp
    thermal_conductivity = 200.0
  []
[]

[Problem]
  type = FEProblem
[]

[Executioner]
  type = Steady
  solve_type = NEWTON
[]

[Postprocessors]
  [output_heat]
    type = SideDiffusiveFluxIntegral
    variable = temp
    boundary = right
    diffusivity = thermal_conductivity
  []
  [output_Tmax]
    type = NodalMaxValue
    variable = temp
  []
  [volume]
    type = VolumePostprocessor
    block = '15 20'
  []
[]

[BCs]
  [rightbc]
    type = DirichletBC
    variable = temp
    boundary = right
    value = 300.0
  []
[]

[Outputs]
  exodus = true
[]
```
The result is 
![image](https://github.com/idaholab/moose/assets/53902666/db5d1f50-8bfc-4eac-9b95-056afae6b2fe)
Tmax does not equal to 1000. 
Why both of these result is quite different from the theory resolution? The error departures from the theory resolution is too high I think. Do I need to change something in [Steady] part? I try set l_tol or nl_rel_tol to fix this, but it seems useless, it only iterates more times and send me the same result......

**Comment by GiudGiud** [comment, marked as answer]
Hello

We can look at that in further depth once you've investigated this fully. For now
```
    ring_intervals = '5'
    num_sectors = 40
```
this is a lot of sectors which do not help to model the radial variation, and not a lot of radial discretization. The sectors help capture the true volume of the cylinder, which is important for the correct definition of the power density variable. How far off is the cylinder volume from the analytical cylinder?

Please up this radial discretization so that the polynomial (quadratic) true shape of the solution is captured better

Guillaume

**Reply by Wang-Yihu** [reply]
Thanks! I refined the 3D mesh and 2D mesh. After enhancing the radial discretization, the result calculated is near the theory resolution. I think you are right!
For 3D mesh, I give my result:
![image](https://github.com/idaholab/moose/assets/53902666/b01ecd17-d7bb-4d68-8f81-59fdbc8535bd)
And for 2D mesh, I give my result:
![image](https://github.com/idaholab/moose/assets/53902666/a02efd59-e4ee-4387-89e6-a0631537b376)
